src/App/Utilities/AppHelpers.cs

```diff
@@ -196,6 +197,55 @@ public static async Task<string> SendListOptions(ContentPage page, SendView send
             return selection;
         }
 
+        public static async Task<string> AccountListOptions(ContentPage page, AccountViewCellViewModel accountViewCell)
+        {
+            var vaultTimeoutService = ServiceContainer.Resolve<IVaultTimeoutService>("vaultTimeoutService");
+            var platformUtilsService = ServiceContainer.Resolve<IPlatformUtilsService>("platformUtilsService");
+
+            var userId = accountViewCell.AccountView.UserId;
+
+            List<string> options;
+            if (await vaultTimeoutService.IsLoggedOutByTimeoutAsync(userId) ||
+                await vaultTimeoutService.ShouldLogOutByTimeoutAsync(userId))
+            {
+                options = new List<string> { AppResources.RemoveAccount };
+            }
+            else if (await vaultTimeoutService.IsLockedAsync(userId) ||
+                     await vaultTimeoutService.ShouldLockAsync(userId))
+            {
+                options = new List<string> { AppResources.LogOut };
+            }
+            else
+            {
+                options = new List<string> { AppResources.Lock, AppResources.LogOut };
+            }
+
+            var accountSummary = accountViewCell.AccountView.Email;
+            if (!string.IsNullOrWhiteSpace(accountViewCell.AccountView.Hostname))
+            {
+                accountSummary += "\n" + accountViewCell.AccountView.Hostname;
+            }
+            var selection = await page.DisplayActionSheet(accountSummary, AppResources.Cancel, null, options.ToArray());
+
+            if (selection == AppResources.Lock)
+            {
+                await vaultTimeoutService.LockAsync(true, true, userId);
+            }
+            else if (selection == AppResources.LogOut || selection == AppResources.RemoveAccount)
+            {
+                var title = selection == AppResources.LogOut ? AppResources.LogOut : AppResources.RemoveAccount;
+                var text = (selection == AppResources.LogOut ? AppResources.LogoutConfirmation
+                    : AppResources.RemoveAccountConfirmation) + "\n\n" + accountSummary;
+                var confirmed =
+                    await platformUtilsService.ShowDialogAsync(text, title, AppResources.Yes, AppResources.Cancel);
+                if (confirmed)
```

Comment:
This could be done in one line. Maybe there is a standard that I don't know and it's preferred to split with a meaningful variable name the result of the method, but in this case I think it's pretty self-explanatory in one line.

Reply:
Saving this for a tech debt exercise as it's an established pattern